pyproject.toml

```diff
@@ -48,3 +48,7 @@ docstring-convention = "google"
 copyright-check = "True"
 copyright-author = "Canonical Ltd."
 copyright-regexp = "Copyright\\s\\d{4}([-,]\\d{4})*\\s+%(author)s"
+
+[tool.pyright]
```

Comment:
I have no strong feeling for one in particular, it's just that I use pyright locally for my LSP.  
Actually I was about to ask if we could add this configuration globally, even if we don't use it in the CI so that it works when used locally.

Reply:
After thinking about it, I decided to remove it. I need to handle this on my side

Reply:
So should we then use `mypy` for this repo as well? Adding type checks is quite valuable and is about to be enforced by `operator-workflows` anyway